packages/block-library/src/gallery/style.scss

```diff
@@ -43,22 +43,45 @@ figure.wp-block-gallery.has-nested-images {
 			width: auto;
 		}
 
+		// Position caption and scrim at the bottom.
+		&:has(figcaption)::before,
 		figcaption {
-			background: linear-gradient(0deg, rgba($color: $black, $alpha: 0.7) 0, rgba($color: $black, $alpha: 0.3) 70%, transparent);
+			position: absolute;
 			bottom: 0;
+			right: 0;
+			left: 0;
+			max-height: 100%;
+		}
+
+		// Create a background blur layer.
+		&:has(figcaption)::before {
+			content: "";
+			height: 100%;
+			max-height: 40%;
+
+			// Blur the background under the gradient scrim.
+			backdrop-filter: blur(3px);
+
+			// Crop the caption so the blur is only on the edge.
+			mask-image: linear-gradient(0deg, rgba($color: $black, $alpha: 1) 20%, rgba($color: $black, $alpha: 0) 100%);
+		}
+
+		// Place the caption at the bottom.
+		figcaption {
 			color: $white;
+			text-shadow: 0 0 1.5px $black;
 			font-size: $default-font-size;
-			left: 0;
-			margin-bottom: 0;
-			max-height: 60%;
+			margin: 0;
 			overflow: auto;
-			padding: 0 8px 8px;
-			position: absolute;
+			padding: 1em;
```

Comment:
This is a non-blocker suggestion.

Because `scrollbar-gutter: stable both-edges` has been applied to the `figcaption`, the padding on the left and right looks wider, we you may want to make it a little smaller.

Below is what it looks like on my environment, but how does it look on yours?

| padding: 1em | padding: 1em 0.5em |
|--------|--------|
| ![image](https://github.com/user-attachments/assets/d3660bc1-261d-42b8-a77a-5a1258fc7909)| ![image](https://github.com/user-attachments/assets/b512376a-3c75-4609-b647-ae07d8fa8d49)|

Reply:
I don't disagree, though since this PR already touches a fair bit of CSS, I wonder if it's not best to do this separately?

Thank you again for the thorough reviews!